app.ts

```diff
@@ -0,0 +1,277 @@
+// Function To Get Number Of Rows That Can Be Displayed While Still Being Readable
+
+const getNoOfRows = () => {
+  const height = window.innerHeight;
+
+  let number = height / 40;
+  let noOfRows = Math.floor(number);
+  return noOfRows;
+};
+
+// Variables
+
+let paramOne = 0;
+let paramTwo = paramOne + getNoOfRows();
+
+//// Functions To Create/Clear The HTML
+
+// Heading Row
+
+const createHeadingRow = (headingData: string) => {
+  const heading: any = document.querySelector("#heading");
+
+  let headings = `<div class="headings" id="headings">${headingData}</div>`;
+  heading.innerHTML += headings;
+};
+
+// Table Content
+
+const createTableContent = (contentData: string) => {
+  const content: any = document.querySelector("#content");
+
+  let table = `<div id="row-${contentData[0]}" class="rows"></div>`;
+  content.innerHTML += table;
+
+  let rows: any = document.querySelector("#row-" + contentData[0] + ".rows");
+  for (let x = 0; x < contentData.length; x++) {
+    let rowCols = `<div class="row_cols">${contentData[x]}</div>`;
+    rows.innerHTML += rowCols;
+  }
+};
+
+// Clear Table Content
+
+const clearTable = () => {
+  const content: any = document.querySelector("#content");
+  const clear = "";
+
+  content.innerHTML = clear;
+};
+
+//// Fetch Requests
+
+// Heading Row (Getting the columns data)
+
+const getHeadings = () => {
+  try {
+    fetch("http://localhost:2050/columns", {
+      method: "GET",
+      headers: { "Content-Type": "application/json" },
+    })
+      .then((res) => res.text())
+      .then((data) => {
+        data = JSON.parse(data);
+        let headingData = data;
+        for (let i = 0; i < headingData.length; i++) {
+          createHeadingRow(headingData[i]);
+        }
+      });
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+// Table Content (Getting the table's data)
+
+const getTable = () => {
+  try {
+    fetch("http://localhost:2050/records?from=" + paramOne + "&to=" + paramTwo, {
+      method: "GET",
+      headers: { "Content-Type": "application/json" },
+    })
+      .then((res) => res.text())
+      .then((data) => {
+        data = JSON.parse(data);
+        let contentData = data;
+        for (let i = 0; i < contentData.length; i++) {
+          createTableContent(contentData[i]);
+        }
+      });
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+// Displays The Current Results Being Shown
+
+const stats = () => {
```

Comment:
In general, this function should not have to do a recordCount API call. The recordCount is constant throughout in this case. You can probably just build your string in here and update it.